hack/e2e.go

```diff
@@ -152,6 +155,15 @@ func main() {
 	log.SetFlags(log.LstdFlags | log.Lshortfile)
 	flag.Parse()
 
+	terminate.Stop()
```

Comment:
These stops don't guarantee that the timer hasn't already fired. Since you have immediate timers above, even if you call stop, there's likely something already waiting to be read off the channel.